fs-cache/src/cache.rs

```diff
@@ -170,45 +175,42 @@ where
             ));
         }
 
-        // First pass: collect metadata only
+        // Collect metadata for all files
         let mut file_metadata = Vec::new();
         for entry in fs::read_dir(&self.path)? {
-            let path = entry?.path();
+            let entry = entry?;
+            let path = entry.path();
             if path.is_file() {
                 let key: K =
                     extract_key_from_file_path(&self.label, &path, true)?;
-                file_metadata.push((key.clone(), self.get_file_size(&key)?));
+                let metadata = entry.metadata()?;
+                let modified = metadata.modified()?;
+                let size = metadata.len() as usize;
+                file_metadata.push((key, size, modified));
             }
         }
 
-        // Sort by size before loading
-        file_metadata.sort_by(|a, b| b.1.cmp(&a.1));
+        // Sort by modified time (most recent first)
```

Comment:
Actually, I'm not sure that pre-loading the most recently modified values would really be beneficial.

We could implement more sophisticated approach with gathering query statistics and recording it somewhere on disk for pre-loading in future. But I would do it in a separate PR and not right now.